CMakeLists.txt

```diff
@@ -152,7 +152,7 @@ set(COMPILER_SUPPORT_LIBRARIES "")
 
 # If we are using MSVC, we need /bigobj.
 if (MSVC)
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /bigobj /Zm400")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /bigobj /Zm400 /std:c++17")
 endif ()
 
 # If we are using MINGW, we need sections and big-obj, otherwise we create too
```

Comment:
@zoq @rcurtin, Yeah, a lot of fun, Even though I have added C++17 flag for Windows, it seems to use the back ported version. Not sure if the MSVC compiler is understanding the `define` macros

Reply:
Indeed, I think this is the issue: https://devblogs.microsoft.com/cppblog/msvc-now-correctly-reports-__cplusplus/

So I guess we need to add `/Zc:__cplusplus` here?

Reply:
Wow, the article is quite interesting.

Reply:
Yeah, it is very interesting and not working at the same time, maybe I need to combine both of them? I do not think so.